GNUmakefile

```diff
@@ -64,109 +64,21 @@ endif
-
-pkg/freebsd_amd64/nomad: $(SOURCE_FILES) ## Build Nomad for freebsd/amd64
-	@echo "==> Building $@..."
-	@CGO_ENABLED=1 GOOS=freebsd GOARCH=amd64 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-pkg/linux_386/nomad: $(SOURCE_FILES) ## Build Nomad for linux/386
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=386 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-pkg/linux_amd64/nomad: $(SOURCE_FILES) ## Build Nomad for linux/amd64
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=amd64 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-pkg/linux_arm/nomad: $(SOURCE_FILES) ## Build Nomad for linux/arm
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm CC=arm-linux-gnueabihf-gcc-5 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-pkg/linux_arm64/nomad: $(SOURCE_FILES) ## Build Nomad for linux/arm64
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=arm64 CC=aarch64-linux-gnu-gcc-5 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-# If CGO support for Windows is ever required, set the following variables
-# in the environment for `go build` for both the windows/amd64 and the
-# windows/386 targets:
-#	CC=i686-w64-mingw32-gcc
-#	CXX=i686-w64-mingw32-g++
-pkg/windows_386/nomad: $(SOURCE_FILES) ## Build Nomad for windows/386
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=windows GOARCH=386 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@.exe"
-
-pkg/windows_amd64/nomad: $(SOURCE_FILES) ## Build Nomad for windows/amd64
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=windows GOARCH=amd64 \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@.exe"
-
-pkg/linux_ppc64le/nomad: $(SOURCE_FILES) ## Build Nomad for linux/arm64
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=ppc64le \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
-
-pkg/linux_s390x/nomad: $(SOURCE_FILES) ## Build Nomad for linux/arm64
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	@CGO_ENABLED=1 GOOS=linux GOARCH=s390x \
-		go build \
-		-trimpath \
-		-ldflags $(GO_LDFLAGS) \
-		-tags "$(GO_TAGS)" \
-		-o "$@"
+	@CGO_ENABLED=1 \
+		GOOS=$(firstword $(subst _, ,$*)) \
+		GOARCH=$(lastword $(subst _, ,$*)) \
+		go build -trimpath -ldflags $(GO_LDFLAGS) -tags "$(GO_TAGS)" -o $(GO_OUT)
+
+pkg/linux_arm/nomad: CC = arm-linux-gnueabihf-gcc-5
+pkg/linux_arm64/nomad: CC = aarch64-linux-gnu-gcc-5
+
+pkg/windows_%/nomad: GO_OUT = $@.exe
+# If CGO support for Windows is ever required, uncomment the following
```

Comment:
I believe this suggestion is no longer meaningful - at least since 2017 when the Makefile was rewritten we've had CGO enabled for Windows targets. In dcdc91fb6b you can kinda see when that flipped.

Reply:
FWIW, I have my doubts whether CGO is actually enabled for Windows. `CGO_ENABLED=1` is set, yet we don't set the windows cross-compile chain values.  On my macOS, I can cross-compile fine with `CGO_ENABLED=1`, but when I actually do add a cgo function, I get:

```
# runtime/cgo
gcc_libinit_windows.c:7:10: fatal error: 'windows.h' file not found
```

Our CI `build-binaries` job (e.g. [1](https://app.circleci.com/pipelines/github/hashicorp/nomad/13853/workflows/8b6cd63d-d23b-417d-9468-013db86337d2/jobs/126597]) builds windows binaries without cross-compilation tools installed.  Though our release container has the x64 arch tools installed according through https://github.com/hashicorp/nomad/blob/v1.0.0/scripts/vagrant-linux-priv-config.sh#L38-L41 .

Reply:
@shoenig I noticed the discrepancy as well; I tend to avoid making changes to existing stuff in a refactor PR. Is it OK to merge this as-is and then make a determination on whether the Windows bit can be removed separately?